### ISSUE TITLE
Moving from lodash to es-toolkit 

### ISSUE THREAD

**Issue by KSJaay** [recharts/recharts#5266]
## Do you want to request a feature or report a bug?
Feature request

## What is the current behavior?
Currently recharts uses lodash, which in my experience takes up about 145kb within my bundle. I've recently been looking at es-toolkit, and its performance/size seems to be much better than lodash. I was planning on raising a PR that moves recharts from lodash to es-toolkit and see how this affects the performance/size.

Would you be fine with me raising that PR?

**Comment by ckifer** [comment]
I would be fine with this. 

I've never heard of es toolkit or used it. Are we positive this is a drop in replacement for everything we're using in recharts? 

There is plenty of usages of lodash that can be probably changed to native alternatives at this point, the worst part is doing that work and making 1000% sure nothing has changed afterwards. 

Let's do it in the 3.x branch and leave 2.x alone.

**Comment by KSJaay** [comment]
Yeah, es toolkit is pretty new and was released earlier this year. It's mainly built to be much faster than lodash and has a MUCH smaller package size. They use a lot of native features, but yeah I agree I think moving to native features would be better than moving to es toolkit.

Where it makes sense, I'll move to native features and test everything. I might leave lodash as it is for now and move to native implementation, that way it should also reduce the final bundle size. In a later PR I'll move to es-toolkit. I think if I do both now the PR is going to be too big.

es-toolkit github: https://github.com/toss/es-toolkit

**Comment by KSJaay** [comment]
Hey, just finished making a few changes and replacing lodash with native code. I've added a list of things I've removed and the code I've replaced it with (basic examples), let me know if you think those are fine and I'll get the PR up later today or tomorrow.

There are some sections that I couldn't replace because the logic behind those functions is rather large/complicated. The only function that isn't complicated but I didn't wanna replace was `isNil` as it checks for both null and undefined, doing this in the code makes it way too messy as I'd have to do `value === null || typeof value === 'undefined'`. So, I was planning on replacing that using es-toolkit instead.

Another issue I ran into was typescript throwing an error when I removed isFunction, it seems like you're checking if a prop is a valid React component using `isFunction`. This can return a false positive, I was looking, and seems like react has `isValidElement` which checks if the passed value is a React/html element. Would you be fine with me using that instead?

Removed: 
- isFunction > typeof value === 'function'
- max > Math.max(...arr)
- min > Math.min(...arr)
- isNan > Number.isNaN
- isNumber > (typeof value === 'number' || value instanceof Number)
- isString > typeof value === 'string'
- omit >
```js 
const newObj = { ...obj};
delete newObj.key;
```
- some > arr.some()
- maxBy > Math.max(...arr)
- sumBy > arr.reduce()
- isObject > typeof value === 'object'
- last > arr[arr.length - 1]
- first > arr[0]
- upperFirst > value.substring(0, 1).toUpperCase() + value.substring(1)
- isBoolean > typeof value === 'boolean'

Need to remove:
- isNil
- isEqual
- range
- get
- throttle
- mapValues
- uniqBy
- isPlainObject

**Comment by ckifer** [comment]
Feel free to make a draft PR and we can go back and forth there. We don't want the changes to be so much that its impossible to review. We can do 2 or more PRs if we need to.

> you're checking if a prop is a valid React component using isFunction

Its not me trust me 😅 theres a lot of legacy stuff in here. 
Can you point me to that code?

**Comment by ckifer** [comment]
`value.substring(0, 1).toUpperCase() + value.substring(1)`

Can you just make an internal utility for this one? Seems appropriate. Also prefer template strings where possible

**Comment by KSJaay** [comment]
Ohh my bad, I read the wrong line. You're using both `React.isValidElement` and `isFunction`. Not sure if it needs the check for isFunction here. I'll update the code and create a new PR, we can talk on there instead.

Also what directory would you like the utility functions? Is [DataUtils](https://github.com/recharts/recharts/blob/0e4d2398f3c6d1db12b5b3ec50d8ddf5a6eb54cd/src/util/DataUtils.ts) fine?

**Code for component check:**
https://github.com/recharts/recharts/blob/573b54ff543fd74a6c2c98a1082503358da828b4/src/component/LabelList.tsx#L88

**Comment by ckifer** [comment]
Yes both of those checks need to be there.

**Comment by ckifer** [comment]
Yeah data utils is fine I guess. If there's a logical grouping of utility functions that doesn't fall into a certain category feel free to make a new file

**Comment by ckifer** [comment]
Moved over to es-toolkit
